version: 3.3"

networks:
  proxy:
    external: true
  default:
    driver: bridge

services:

  nextcloud-redis:
    image: redis:alpine
    container_name: nextcloud-redis
    hostname: nextcloud-redis
    environment:
      REDIS_PASSWORD: ${RANDOM_PASSWORD}
    restart: always

  nextcloud-db:
    image: mariadb
    container_name: nextcloud-db
    command: --transaction-isolation=READ-COMMITTED --log-bin=ROW
    restart: always
    volumes:
      - ~/homelab_data/nextcloud/database:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=${RANDOM_PASSWORD}
      - MYSQL_PASSWORD=${NEXTCLOUD_DB_PASSWORD}
      - MYSQL_USER=${NEXCLOUD_DB_USER}
      - MYSQL_DATABASE=nextcloud
      - MYSQL_INITDB_SKIP_TZINFO=1 # Fix db startup issues

  nextcloud-app:
    image: nextcloud
    container_name: nextcloud-app
    restart: always
    depends_on:
      - nextcloud-db
      - nextcloud-redis
    environment:
        REDIS_HOST: nextcloud-redis
        REDIS_HOST_PASSWORD: ${RANDOM_PASSWORD}
    volumes:
      - ~/homelab_data/nextcloud/app:/var/www/html
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.jupyter.entrypoints=http"
      - "traefik.http.routers.jupyter.rule=Host(`nextcloud.${DOMAIN}`)"
      - "traefik.http.services.jupyter.loadbalancer.server.port=80"
    networks:
      - proxy
